lua/toggleterm.lua

```diff
@@ -352,22 +352,23 @@ local function setup_autocommands(_)
     end
 
     -- Extra mouse fix for tmux
-    -- If tmux mouse mode is enabled
-    local output = vim.fn.system 'tmux display -p "#{mouse}"'
-    if output:sub(1, 1) == "1" then
-      -- Disable tmux mouse while using toggleterm
-      api.nvim_create_autocmd({ "TermEnter", "WinEnter <buffer>" }, {
-        desc = "Disable tmux mouse while using toggleterm",
-        group = AUGROUP,
-        callback = function() vim.fn.system "tmux set mouse off" end,
-      })
-
-      -- Enable tmux mouse when mouse leaves toggleterm
-      api.nvim_create_autocmd({ "WinLeave <buffer>" }, {
-        desc = "Enable tmux mouse when mouse leaves toggleterm",
-        group = AUGROUP,
-        callback = function() vim.fn.system "tmux set mouse on" end,
-      })
+    if os.getenv "TMUX" ~= nil then
+      local output = vim.fn.system 'tmux display -p "#{mouse}"'
+      if output:sub(1, 1) == "1" then
+        -- Disable tmux mouse while using toggleterm
+        autocmd({ "TermEnter", "WinEnter <buffer>" }, {
```

Comment:
If an autocommand is buffer local it should pass in the buffer as an argument to the function not using `<buffer>`. I think this should be a separate autocmd with `buffer = num`

Reply:
Running the same code without "WinEnter <buffer>" achieves the same, but only after moving the mouse at least once. So the event seems to be working as it should.

I've tried removing the <buffer> part as [it does not appear in the documentation](https://neovim.io/doc/user/autocmd.html#WinEnter) and still seems to work as expected. So it is probably ignored by neovim for better backwards compatibility. I'm gonna leave it as just "WinEnter".

Also, you probablly saw it but the most important part of the code is 

``` lua
if vim.bo.filetype == "toggleterm" then
```
Meaning wathever we do is only gonna affect toggleterm.